ros/src/tl_detector/tl_detector.py

```diff
@@ -272,6 +268,22 @@ def get_light_state(self, light):
         # Get classification
         return self.light_classifier.get_classification(img_traffic_light)
 
+    def get_tl_waypoints(self):
+        """ Converts list self.config['light_positions'] with trafic
+        light positions to get_closest_waypoint array with PoseStamped
+        traffic light waypoints
+
+        """
+
+        for light_position in self.config['light_positions']:
```

Comment:
We should use `self.lights`, which contains the `z` coordinate which is needed for cropping the image. There was a discussion on Slack about that, and that information *will* be available always, both in the simulator and in the test track (without the state of the light).

Reply:
Thats pretty confusing because they have written explicitly in source code:
        '''
        /vehicle/traffic_lights helps you acquire an accurate ground truth
        data source for the traffic light classifier, providing the location
        and current color state of all traffic lights in the simulator.
        This state can be used to generate classified images or subbed into
        your solution to help you work on another single component of the node.
        This topic won't be available when testing your solution in real life
        so don't rely on it in the final submission.
        '''

Are you sure about that?

Reply:
I got this on Slack:

Carlos Galvez [4 days ago] 
@caleb Thanks! So if I understand correctly, we should always subscribe to `/vehicle/traffic_lights`. The only difference between simulator and real life, is that we will *not* get the status of the light in the test track. Is that correct? In other words, we will *always* get `(x, y, z, yaw)` of the traffic light both in test track and simulator?


Caleb Kirksey [3 days ago] 
@carlosgalvezp Yes, exactly

Reply:
Yes I saw it now, but he hasn't answered to your last question, I have asked him one more time.

Reply:
Great, thanks! I've added a Github issue for them as well to put more pressure :)